tst/shaka_scheme/system/base/unit-Environment.cpp

```diff
@@ -0,0 +1,200 @@
+
+TEST(EnvironmentUnitTest, parent_constructor) {
+  // Given: an environment with parent as null pointer.
+  Environment e(nullptr);
+
+  // When: you try to get the parent
+  // Then: it will return a null pointer.
+  ASSERT_EQ(e.get_parent(), nullptr);
+
+  // Given: a pointer to a parent Environment
+  //        an Environment with parent as pointer to parent Environment.
+  auto parent = std::make_shared<Environment>(nullptr);
+  e.set_parent(parent);
+
+  // When: you try to get the parent.
+  // Then: it will return the pointer to the parent Environment.
+  ASSERT_EQ(e.get_parent(), parent);
+}
+
+TEST(EnvironmentUnitTest, get_parent) {
+  // Given: a pointer to a parent Environment
+  //        environment with parent as null pointer.
+  Environment e(nullptr);
+  auto parent = std::make_shared<Environment>(nullptr);
+
+  // When: you try to get the parent
+  // Then: it will return the pointer to the null pointer.
+  ASSERT_EQ(e.get_parent(), nullptr);
+
+  // Given: environment with parent set to pointer to parent Environment.
+  e.set_parent(parent);
+
+  // When: you try to get the parent
+  // Then: it will return the pointer to the parent Environment.
+  ASSERT_EQ(e.get_parent(), parent);
+
+}
+
+TEST(EnvironmentUnitTest, set_parent) {
+  // Given: a pointer to a parent Environment.
+  //        a child Environment with parent initialized to null pointer.
+  auto parent = std::make_shared<Environment>(nullptr);
+  Environment child(nullptr);
+
+  // When: you set the child's parent to the parent pointer.
+  child.set_parent(parent);
+
+  // Then: the child's parent will be changed to the parent pointer.
+  ASSERT_EQ(child.get_parent(), parent);
+}
+
+TEST(EnvironmentUnitTest, set_value) {
+  // Given: an environment with parent initialized to the null pointer.
+  //        a Symbol
+  //        a Value
+  Environment e(nullptr);
+  Symbol key("x");
+  auto value = create_node(String("test"));
+
+  // When: you set the key to the specified value
+  e.set_value(key, value);
+
+  // Then: the key will have the correct value
+  ASSERT_EQ(e.get_value(key), value);
+}
+TEST(EnvironmentUnitTest, get_value_success) {
+  // Given: an environment, key and value
+  Environment e(nullptr);
+  Symbol key1("x");
+  auto value1 = create_node(String("test"));
+
+  // When: you set the key to the specified value
+  e.set_value(key1, value1);
+
+  // Then: the correct value will be set
+  ASSERT_EQ(e.get_value(key1), value1);
+
+  // Given: an environment with parent set to pointer to parent environment,
+  //        key and value
+  auto parent = std::make_shared<Environment>(nullptr);
+  e.set_parent(parent);
+  Symbol key2("y");
+  auto value2 = create_node(String("parent test"));
+
+  // When: you set the key to the specified value in only the parent
+  parent->set_value(key2, value2);
+
+  // Then: the correct value will be returned
+  ASSERT_EQ(e.get_value(key2), value2);
+}
+TEST(EnvironmentUnitTest, get_value_invalid_key_failure) {
```

Comment:
I see that you already changed the implementation correctly if the unit test case passes. Therefore, just change the documentation on the function.